lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -329,11 +332,44 @@ legacyUtxoWitness xpub bytes = TxWitness $ BL.toStrict $ runPut $ do
     putByteString (unXPub xpub)
     putByteString bytes
 
+-- | Decode the contents of a delegated transaction @Transaction@-message.
+getDelegatedCertificateTransaction :: Int -> Get (Tx, [TxWitness])
+getDelegatedCertificateTransaction =
+    getGenericTransaction MsgTypeCertificate "getDelegatedCertificateTransaction"
+
+stakeCertificate
+    :: PoolId
+    -> XPub
+    -> Put
+stakeCertificate (PoolId poolId) xpub = do
```

Comment:
My guess is that we should have just probably `StakeCertificate` which is constructed elsewhere basing on `PoolId` and `XPub` and is signed using corresponding `XPrv`. And we should just `putByteString` here

Reply:
Actually not, we need to know whether a certificate belongs to us by looking at which `XPub` (reward account) it is attached. We could use some sort of opaque type like addresses, but It's probably clearer to represent certificate as a link between `PoolId` and `XPub` at the moment.

Reply:
ok, make sense. Added `PoolId` and 64-bytes `ByteString` after regular tx payload.

Reply:
This should probably be called `putStakeCertificate` for consistency. Also, looking at the rust implementation, it seems that the account id is inserted first in the serialization (https://github.com/input-output-hk/chain-libs/blob/master/chain-impl-mockchain/src/certificate/delegation.rs#L57-L60) although, our unit tests should catch that!

Reply:
and I see accountId has length of 32 : https://github.com/input-output-hk/chain-libs/blob/master/chain-impl-mockchain/src/certificate/delegation.rs#L66